week1/homework/src/server.js

```diff
@@ -5,11 +5,43 @@ const http = require('http');
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
+let responseObject = {'state': 10};
+const responseError = { "error": "Not found"};
+const urls = ['/state', '/add', '/subtract', '/reset'];
 function createServer(port) {
-  let state = 10;
-
+  let responseObject = {'state': 10};
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    const status = (urls.some(elm => elm === request.url))? 200 : 404;
+    response.writeHead(status, {"Content-type": "application/json"})
+    switch(request.url)
+  	{        
+      case '/add':
+      {
+        responseObject.state = responseObject.state + 1;
+        break;
+      }
+      case '/subtract':
+      {
+        responseObject.state = responseObject.state - 1;
+        break;
+      }
+      case '/reset':
+      {
+        responseObject.state = 10;
+        break;
+      }
+      default:
+        break;
+    }
+    
+    if(urls.some(elm => elm === request.url))
+      response.write(JSON.stringify(responseObject, null, 2));
+    else
+      response.write(JSON.stringify(responseError, null, 2));
```

Comment:
This is the third place you've put (a part of the) logic to handle the error.

You can put the response.write directly in the `switch`.

`JSON.stringify(responseError)` is valide aswel.